### PR TITLE
Add indice, run length helper and tests

### PR DESCRIPTION
**Pull request by aulemahal**
### Pull Request Checklist:
- [ ] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #xyz
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
- [ ] `bumpversion (minor / major / patch)` has been called on this branch
- [ ] Tags have been pushed (`git push --tags`)

* **What kind of change does this PR introduce?** 
Adds `first_day_below` indice and indicator. Also `run_length.first_run_after_date`.

* **Does this PR introduce a breaking change?** 
Nein

* **Other information**:
It will be mainly used to compute the first frost day. I did not name it `first_frost_day` so its more general. However, that makes it in contradiction with `last_frost_day`...

### CODE REVIEW

**Comment by aulemahal** [comment]
The difference is that it doesn't check for a preexisting run. Temperature doesn't have to be above thresh before "date". It's literally a `first_run` with data before the given date masked. I tried to use `run_end_after_date` with the opposite condition, as you suggest), but in addition to my first point, there was also the problem that `run_end_after_date` (or `growing_season_end` will return the last time if the run doesn't end within the timeseries.

File: HISTORY.rst [hunk 1]
```diff
@@ -4,6 +4,7 @@ History
 
 0.17.x (2020-05-15)
 -------------------
+* Added indicator `first_day_below` and run length helper `first_run_after_date`
```
Comment (Zeitsperre) [inline HISTORY.rst:7]:
```suggestion
0.18.x
------
* Added indicator `first_day_below` and run length helper `first_run_after_date`

0.17.0 (2020-05-15)
-------------------
```

File: xclim/indices/run_length.py [hunk 2]
```diff
@@ -388,6 +388,46 @@ def run_end_after_date(
     return end.where(beg.notnull())
 
 
+def first_run_after_date(
+    da: xr.DataArray,
+    window: int,
+    date: str = "07-01",
+    dim: str = "time",
+    coord: str = "dayofyear",
```
Comment (Zeitsperre) [inline xclim/indices/run_length.py:396]:
```suggestion
    coord: Opitonal[str] = "dayofyear",
```

File: HISTORY.rst (hunk 1 above)
Comment (aulemahal) [reply HISTORY.rst:7]: woups, thanks!

File: xclim/indices/run_length.py (hunk 2 above)
Comment (aulemahal) [reply xclim/indices/run_length.py:396]: haha you sure?

Comment (Zeitsperre) [reply xclim/indices/run_length.py:396]: It's marked optional in the docstring. Both can't be right?

Comment (aulemahal) [reply xclim/indices/run_length.py]: oui oui you're right, I meant for the "opitonal" ;)

Comment (Zeitsperre) [reply xclim/indices/run_length.py]: Oof. That's embarrassing.

**Pull request merged**